rfc-drafts/stream.md

```diff
@@ -233,19 +247,62 @@ Designing such a migration feature is out of scope for this RFC.
 
 ## Async iteration syntax
 
-We may wish to introduce some dedicated syntax, analogous to `for` 
+Currently, if someone wishes to iterate over a `Stream` as defined in the `futures` crate,
+they are not able to use  `for` loops, they must use `while let` and `next/try_next` instead.
+
+We may wish to extend the `for` loop so that it works over streams as well. 
+
+```rust
+#[async]
+for elem in stream { ... }
+```
+
+Designing this extension is out of scope for this RFC.
+
+## "Attached" streams
+
+There has been much discussion around attached/detached streams.
+
+### Definitions
+
+[Source](https://smallcultfollowing.com/babysteps/blog/2019/12/10/async-interview-2-cramertj-part-2/#the-need-for-streaming-streams-and-iterators)
+
+In a **detached** stream, the `Item` that gets returned by `Stream` is "detached" from self. This means it can be stored and moved about independently from `self`.
+
+In an **attached** stream, the `Item` that gets returned by `Stream` may be borrowed from `self`. It can only be used as long as the `self` reference remains live.
+
+### Alternate terminology
+
+An alternative terminology - which may be easier to understand in terms of ownership and borrowed - involves **owned** and **borrowed** streams.
```

Comment:
Probably we should settle on one terminology before publishing the RFC...I don't like the term "owned" tream because the question is not so much about *ownership* as *who is the owner* -- i.e., is it the stream or the consumer of the stream? And that is not clarified.

Reply:
True - I'm wondering if attached/detached might be the best terminology. It is not perfect, but it seems like the clearest.

Reply:
I guess one advantage of attached/detached is that it works both ways -- i.e., an "attached" stream means that the *stream and the items it producs are attached to one another*.

Reply:
Very true!